openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/OpenAPIComponentMapper.java

```diff
@@ -719,54 +719,100 @@ private ArraySchema handleArray(int arrayDimensions, Schema property, ArraySchem
+        properties.setMaximum(maximum);
+        return properties;
+    }
 
-        try {
-            BigDecimal minimum = null;
-            BigDecimal maximum = null;
-            if (constraintAnnot.getMinValue().isPresent()) {
-                try {
-                    minimum = BigDecimal.valueOf(Integer.parseInt(constraintAnnot.getMinValue().get()));
-                } catch (NumberFormatException e) {
-                    minimum = BigDecimal.valueOf(NumberFormat.getInstance().parse(
-                            constraintAnnot.getMinValue().get()).doubleValue());
-                }
-            }
+    /**
+     * This util uses to set the string constraint values for relevant schema field.
+     */
+    private Schema setStringConstraintValuesToSchema(ConstraintAnnotation constraintAnnot, Schema properties) {
+        properties.setMaxLength(constraintAnnot.getMaxLength().isPresent() ?
+                Integer.valueOf(constraintAnnot.getMaxLength().get()) : null);
+        properties.setMinLength(constraintAnnot.getMinLength().isPresent() ?
+                Integer.valueOf(constraintAnnot.getMinLength().get()) : null);
+        return properties;
+    }
 
-            if (constraintAnnot.getMaxValue().isPresent()) {
-                try {
-                    maximum = BigDecimal.valueOf(Integer.parseInt(constraintAnnot.getMaxValue().get()));
-                } catch (NumberFormatException e) {
-                    maximum = BigDecimal.valueOf((NumberFormat.getInstance()
-                                    .parse(constraintAnnot.getMaxValue().get()).doubleValue()));
-                }
+    /**
+     * This util uses to set the array constraint values for relevant schema field.
+     */
+    private Schema setArrayConstraintValuesToSchema(ConstraintAnnotation constraintAnnot, Schema properties) {
+        properties.setMaxItems(constraintAnnot.getMaxLength().isPresent() ?
+                Integer.valueOf(constraintAnnot.getMaxLength().get()) : null);
+        properties.setMinItems(constraintAnnot.getMinLength().isPresent() ?
+                Integer.valueOf(constraintAnnot.getMinLength().get()) : null);
+        return properties;
+    }
 
+    /**
+     * This util uses to set the constraint values for relevant schema field.
+     */
+    private Schema setConstraintValueToSchema(ConstraintAnnotation constraintAnnot, Schema properties) {
+        try {
+            if (properties instanceof ArraySchema) {
+                setArrayConstraintValuesToSchema(constraintAnnot, properties);
+            } else if (properties instanceof StringSchema){
+                setStringConstraintValuesToSchema(constraintAnnot, properties);
+            } else if (properties instanceof IntegerSchema) {
+                setIntegerConstraintValuesToSchema(constraintAnnot, properties);
+            } else if (properties instanceof NumberSchema) {
+                setNumberConstraintValuesToSchema(constraintAnnot, properties);
```

Comment:
If the `properties` variable is not an instance of any of the expected schema types, it will just return. Is this intended? Or should there be any error handling?

Reply:
Only the instances of expected types will call this method (Int, Number[Float, Double], String, Array).  Since this `setConstraintValueToSchema()` method acts as a setter, I have changed the return type of it as void and made all other necessary changes related to that as well. Now it will be more clear.